rmf_human_detector/src/HumanDetector.cpp

```diff
@@ -0,0 +1,217 @@
+#include <filesystem>
+#include <rclcpp/wait_for_message.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <rmf_human_detector/HumanDetector.hpp>
+
+
+HumanDetector::HumanDetector()
+: Node("human_detector"), _data(std::make_shared<Data>())
+{
+  make_detector();
+
+  _data->_obstacles_pub = this->create_publisher<Obstacles>(
+    "/rmf_obstacles",
+    rclcpp::SensorDataQoS()
+  );
+
+  _data->_image_detections_pub =
+    this->create_publisher<sensor_msgs::msg::Image>(
+    _data->_image_detections_topic,
+    rclcpp::QoS(10).reliable()
+    );
+
+  _data->_image_sub = this->create_subscription<sensor_msgs::msg::Image>(
+    _data->_camera_image_topic,
+    rclcpp::SensorDataQoS(),
+    [=](const sensor_msgs::msg::Image::ConstSharedPtr& msg)
```

Comment:
Why are you capturing `=`? The whole point of the data ptr strategy was to provide a succinct way to capture everything you need by copy without capturing everything. Avoid these kind of global captures. Update code below to use `data` 

```suggestion
    [data = _data](const sensor_msgs::msg::Image::ConstSharedPtr& msg)
```